pkg/manager/manager_test.go

```diff
@@ -50,93 +57,161 @@ const (
-	change1, err = change.NewChange([]*devicechangetypes.ChangeValue{
-		config1Value01, config1Value02, config1Value03}, "Original Config for test switch")
-	if err != nil {
-		log.Error(err)
-		os.Exit(-1)
-	}
-	changeStoreTest = make(map[string]*change.Change)
-	changeStoreTest[store.B64(change1.ID)] = change1
 
-	device1config, err = store.NewConfiguration("Device1", "1.0.0", "TestDevice",
-		[]change.ID{change1.ID})
-	if err != nil {
-		log.Error(err)
-		os.Exit(-1)
+	ctrl := gomock.NewController(t)
+
+	mockDeviceCache := devicestore.NewMockCache(ctrl)
+
+	// Data for default configuration
+
+	change1 := types.Change{
+		Values: []*types.ChangeValue{
+			config1Value01, config1Value02, config1Value03},
+		DeviceID:      Device1,
+		DeviceVersion: DeviceVersion,
+	}
+	deviceChange1 := &devicechange.DeviceChange{
+		Change: &change1,
+		ID:     "Change1",
+		Status: typeschange.Status{State: typeschange.State_COMPLETE},
 	}
 
-	configurationStoreTest = make(map[store.ConfigName]store.Configuration)
-	configurationStoreTest[device1config.Name] = *device1config
+	now := time.Now()
+	networkChange1 := &network.NetworkChange{
+		ID:      NetworkChange1,
+		Changes: []*types.Change{&change1},
+		Updated: now,
+		Created: now,
+		Status:  typeschange.Status{State: typeschange.State_COMPLETE},
+	}
 
-	ctrl := gomock.NewController(t)
+	// Mocks for stores
 
+	// Mock Leadership Store
 	mockLeadershipStore := mockstore.NewMockLeadershipStore(ctrl)
-	mockMastershipStore := mockstore.NewMockMastershipStore(ctrl)
-	mockNetworkChangesStore := mockstore.NewMockNetworkChangesStore(ctrl)
-	mockDeviceChangesStore := mockstore.NewMockDeviceChangesStore(ctrl)
-	mockDeviceStore := mockstore.NewMockDeviceStore(ctrl)
-	mockNetworkSnapshotStore := mockstore.NewMockNetworkSnapshotStore(ctrl)
-	mockDeviceSnapshotStore := mockstore.NewMockDeviceSnapshotStore(ctrl)
-	mockDeviceCache := devicestore.NewMockCache(ctrl)
-
 	mockLeadershipStore.EXPECT().Watch(gomock.Any()).AnyTimes()
 	mockLeadershipStore.EXPECT().IsLeader().AnyTimes()
+
+	// Mock Mastership Store
+	mockMastershipStore := mockstore.NewMockMastershipStore(ctrl)
 	mockMastershipStore.EXPECT().Watch(gomock.Any(), gomock.Any()).AnyTimes()
+
+	// Mock Network changes store
+	networkChangesList := make([]*network.NetworkChange, 0)
+	mockNetworkChangesStore := mockstore.NewMockNetworkChangesStore(ctrl)
 	mockNetworkChangesStore.EXPECT().Watch(gomock.Any()).AnyTimes()
+	mockNetworkChangesStore.EXPECT().Create(gomock.Any()).DoAndReturn(
+		func(networkChange *networkchangetypes.NetworkChange) error {
+			networkChangesList = append(networkChangesList, networkChange)
+			return nil
+		}).AnyTimes()
+	mockNetworkChangesStore.EXPECT().Get(gomock.Any()).DoAndReturn(
+		func(id networkchangetypes.ID) (*networkchangetypes.NetworkChange, error) {
+			for _, networkChange := range networkChangesList {
+				if networkChange.ID == id {
+					return networkChange, nil
+				}
```

Comment:
you can avoid this 'if' by passing the ID in the .Get() at line 126, basically it will fail if it receives anything other that that ID.
same applies for all other calls.

Reply:
The Create() method can dynamically update the mock. You can't use static mocks here as the tests modify the store as they run.